.github/workflows/openwrt.yml

```diff
@@ -1,112 +1,103 @@
+      matrix:
+        include:
+          - arch: arc_archs
+            target: archs38-generic
+            runtime_test: false
+
+          - arch: arm_cortex-a9_vfpv3-d16
+            target: mvebu-cortexa9
+            runtime_test: false
+
+          - arch: mips_24kc
+            target: ath79-generic
+            runtime_test: false
+
+          - arch: mipsel_24kc
+            target: mt7621
+            runtime_test: false
+
+          - arch: powerpc_464fp
+            target: apm821xx-nand
+            runtime_test: false
+
+          - arch: powerpc_8540
+            target: mpc85xx-p1010
+            runtime_test: false
+
+          - arch: aarch64_cortex-a53
+            target: mvebu-cortexa53
+            runtime_test: true
+
+          - arch: arm_cortex-a15_neon-vfpv4
+            target: armvirt-32
+            runtime_test: true
+
+          - arch: i386_pentium-mmx
+            target: x86-geode
+            runtime_test: true
+
+          - arch: x86_64
+            target: x86-64
+            runtime_test: true
 
     steps:
-      - name: Checkout openwrt
-        uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
-          path: openwrt
-          repository: openwrt/openwrt
-
-      - name: Set openwrt ref
-        id: openwrt_ref
-        run: |
-          echo ::set-output name=REF::$(git rev-parse --short HEAD)
-          git rev-parse --short HEAD
-
-      - name: Checkout n2n
-        uses: actions/checkout@v2
-        with:
-          path: n2n
           fetch-depth: 0
 
-      - name: Fix Checkout
-        run: |
-          git fetch --force --tags
-        working-directory: n2n
-
-      - name: Set n2n ref
-        id: n2n_ref
-        run: |
-          echo ::set-output name=REF::$(./scripts/version.sh)
-          ./scripts/version.sh
-        working-directory: n2n
+      - name: Build
+        uses: openwrt/gh-action-sdk@v5
+        env:
+          ARCH: ${{ matrix.arch }}
+          FEEDNAME: n2n_ci
+          PACKAGES: openwrt
 
-      - name: Copy n2n package definition into openwrt
-        run: |
-          cp -r n2n/packages/openwrt openwrt/package/n2n
-        working-directory: ./
+      - name: Move created packages to project dir
+        run: cp bin/packages/${{ matrix.arch }}/n2n_ci/*.ipk . || true
 
-      - name: Cache openwrt source downloads
-        uses: actions/cache@v2
+      - name: Store packages
+        uses: actions/upload-artifact@v2
         with:
-          path: openwrt/dl
-          key: openwrt-dl-${{ steps.openwrt_ref.outputs.REF }}
-
-      - name: Setup openwrt config and environment
-        run: |
-          echo "CONFIG_TARGET_x86=y" >.config
-          echo "CONFIG_TARGET_x86_64=y" >>.config
-
-      - name: Add n2n package to openwrt config
-        run: |
-          echo "CONFIG_PACKAGE_n2n-edge=m" >>.config
-          echo "CONFIG_PACKAGE_n2n-supernode=m" >>.config
+          name: ${{ matrix.arch}}-${{ github.sha}}-packages
```

Comment:
Uploading each different build to a different artifact container makes it hard to see the artifacts.  Have you considered creating one artifact for "packages-ipkg", just like we are doing with the RPM and dpkg builds?

Reply:
I'm not familiar with `actions/upload-artifact`. Can it be called multiple times with the same name?